test/widgets/topic_list_test.dart

```diff
@@ -0,0 +1,375 @@
+
+      check(pushedRoutes).single
+        .isA<MaterialAccountWidgetRoute>()
+        .page.isA<MessageListPage>()
+        .initNarrow.equals(TopicNarrow(stream.streamId, topic.name));
+    });
+
+    testWidgets('shows unread count badge', (tester) async {
+      final stream = eg.stream();
+      final topic = eg.getStreamTopicsEntry(name: 'test topic');
+
+      await setupTopicListPage(tester,
+        stream: stream,
+        topics: [topic],
+        unreadMsgs: eg.unreadMsgs(channels: [
+          eg.unreadChannelMsgs(
+            streamId: stream.streamId,
+            topic: 'test topic',
+            unreadMessageIds: [1, 2, 3]),
+        ]));
+      await tester.pumpAndSettle();
+
+      check(find.text('3')).findsOne();
+    });
+
+    testWidgets('shows channel name in app bar', (tester) async {
+      final stream = eg.stream(name: 'Test Stream');
+      await setupTopicListPage(tester,
+        stream: stream,
+        topics: [],
+        subscription: eg.subscription(stream));
+      await tester.pumpAndSettle();
+
+      check(find.text('Test Stream')).findsOne();
+    });
+
+    testWidgets('shows channel feed button in app bar', (tester) async {
+      final stream = eg.stream();
+
+      await setupTopicListPage(tester,
+        stream: stream,
+        topics: [],
+        subscription: eg.subscription(stream));
+      await tester.pumpAndSettle();
+
+      check(find.byIcon(ZulipIcons.message_feed)).findsOne();
+    });
+
+    testWidgets('navigates to channel narrow on channel feed button tap', (tester) async {
+      final pushedRoutes = <Route<void>>[];
+      final navObserver = TestNavigatorObserver()
+        ..onPushed = (route, prevRoute) => pushedRoutes.add(route);
+      final stream = eg.stream();
+      final message = eg.streamMessage(stream: stream);
+
+      await setupTopicListPage(tester,
+        stream: stream,
+        topics: [],
+        subscription: eg.subscription(stream),
+        navObservers: [navObserver]);
+
+      final store = await testBinding.globalStore.perAccount(eg.selfAccount.id);
+      await store.addUser(eg.selfUser);
+
+      final connection = store.connection as FakeApiConnection;
+      connection.prepare(json: eg.newestGetMessagesResult(
+        foundOldest: true,
+        messages: [message],
+      ).toJson());
+
+      await tester.pumpAndSettle();
+      pushedRoutes.clear();
+
+      await tester.tap(find.byIcon(ZulipIcons.message_feed));
+      await tester.pumpAndSettle();
+
+      check(pushedRoutes).single
+        .isA<MaterialAccountWidgetRoute>()
+        .page.isA<MessageListPage>()
+        .initNarrow.equals(ChannelNarrow(stream.streamId));
+    });
+
+    bool hasIcon(WidgetTester tester, {
+      required Widget? parent,
+      required IconData icon,
+    }) {
+      check(parent).isNotNull();
+      return tester.widgetList(find.descendant(
+        of: find.byWidget(parent!),
+        matching: find.byIcon(icon),
+      )).isNotEmpty;
```

Comment:
It seems a bit unnecessary to convert the finder result to a `bool`. Let's just return the finder from this helper (we should then rename this too), and use `check(…).findsOne` etc on the result as appropriate.

This way we have more information about the find result when the check fails.